### PR TITLE
feat: errorCode from execution output

### PR DESCRIPTION
**Pull request by duvanmonsa**
Passing errorCode from execution runner before saving the file.

### CODE REVIEW

**Comment by duvanmonsa** [comment]
https://github.com/stateful/vscode-runme/issues/776

File: src/client/components/terminal/index.ts [hunk 1]
```diff
@@ -305,6 +305,7 @@ export class TerminalView extends LitElement {
   protected windowSize: IWindowSize
 
   protected rows: number = 10
+  protected errorCode?: any
```
Comment (christian-bromann) [inline src/client/components/terminal/index.ts:308]: We know the type of this one right?
```suggestion
  protected errorCode?: number
```

File: src/client/components/terminal/index.ts [hunk 2]
```diff
@@ -305,6 +305,7 @@ export class TerminalView extends LitElement {
   protected windowSize: IWindowSize
 
   protected rows: number = 10
+  protected errorCode?: number | void
```
Comment (christian-bromann) [inline src/client/components/terminal/index.ts:308]: This doesn't work?
```suggestion
  protected errorCode?: number
```

Comment (duvanmonsa) [reply src/client/components/terminal/index.ts:308]: @christian-bromann nop I tried it but it was not working where we get the errorCode returns that type I can for the type if you want. 
image

Comment (christian-bromann) [reply src/client/components/terminal/index.ts:308]: ok, I think it is fine then to keep as is.

**Review by degrammer — APPROVED** [review]
LGTM, despite this approach works, let's check with @sourishkrout  about this.

**Comment by sourishkrout** [comment]
> Passing errorCode from execution runner before saving the file.

What GitHub issue is this solving? Just for context.

**Comment by admc** [comment]
[I think it's this](https://github.com/stateful/cloud/issues/117)

**Comment by duvanmonsa** [comment]
@sourishkrout my bad yes this one https://github.com/stateful/vscode-runme/issues/776

**Comment by duvanmonsa** [comment]
> [I think it's this](https://github.com/stateful/cloud/issues/117)

that one is different @admc that's on the FE for cloud.

**Comment by sourishkrout** [comment]
> @sourishkrout my bad this one #776

This is definitely just a bandaid so depending on when we can expect a full resolution of #776 (no longer relying on terminal state) we can decided whether or not to merge. What prompted this intermediate fix? Unless there's a good reason I feel like we should fix this once and for all instead of spending time on surface-level bandaids.

**Comment by sourishkrout** [comment]
> > [I think it's this](https://github.com/stateful/cloud/issues/117)
> 
> that one is different @admc that's on the FE for cloud.

It's likely both.

**Comment by duvanmonsa** [comment]
Then maybe we don't know how to solve it correctly, Right now we are getting the errorCode directly from the response of the execution and it was working as the expected behavior.  I would need some help to see how we can solve this in the proper way.

**Comment by degrammer** [comment]
Closing in favor of https://github.com/stateful/vscode-runme/pull/829

**Pull request closed without merging**